lib/plottr_import_export/src/importer/scrivener/__tests__/importer.test.js

```diff
@@ -0,0 +1,911 @@
+    const newNote1 = createNewNote(currentState.notes, { title: TEST_STRING_1, bookId })
+    const newNote2 = createNewNote(currentState.notes, { title: TEST_STRING_2, bookId })
+    const newNote3 = createNewNote(currentState.notes, { title: TEST_STRING_3, bookId })
+    const newNote4 = createNewNote(currentState.notes, { title: TEST_STRING_3, bookId })
+
+    it('should return an id interger', () => {
+      expect(typeof newNote1).toBe('number')
+      expect(typeof newNote2).toBe('number')
+    })
+
+    it('should return an incremented value by 1', () => {
+      expect(newNote1).toBe(1)
+      expect(newNote2).toBe(2)
+      expect(newNote3).toBe(3)
+    })
+
+    it('should return found note id if title of note exists', () => {
+      expect(newNote4).toEqual(newNote3)
+    })
+  })
+
+  describe('createNewPlace', () => {
+    const newPlace1 = createNewPlace(currentState.places, { name: TEST_STRING_1, bookId })
+    const newPlace2 = createNewPlace(currentState.places, { name: TEST_STRING_2, bookId })
+    const newPlace3 = createNewPlace(currentState.places, { name: TEST_STRING_3, bookId })
+    const newPlace4 = createNewPlace(currentState.places, { name: TEST_STRING_3, bookId })
+    it('should return an id interger', () => {
+      expect(typeof newPlace1).toBe('number')
+      expect(typeof newPlace2).toBe('number')
+    })
+
+    it('should return an incremented value by 1', () => {
+      expect(newPlace1).toBe(1)
+      expect(newPlace2).toBe(2)
+      expect(newPlace3).toBe(3)
+    })
+
+    it('should return found place id if name of place exists', () => {
+      expect(newPlace4).toEqual(newPlace3)
+    })
+  })
+
+  describe('createNewCard', () => {
+    let currentCards = currentState.cards
+    currentCards = createNewCard(currentCards, { title: TEST_STRING_1, bookId, id: 1 })
+    currentCards = createNewCard(currentCards, { title: TEST_STRING_2, bookId, id: 2 })
+    currentCards = createNewCard(currentCards, { title: TEST_STRING_3, bookId, id: 3 })
+    currentCards = createNewCard(currentCards, { title: TEST_STRING_3, bookId, id: 4 })
+    currentCards = createNewCard(currentCards, { title: TEST_STRING_3, bookId, id: 4 })
+    const similarCards = currentCards.filter((item) => item.title == TEST_STRING_3)
+
+    it('should return an Array', () => {
+      expect(Array.isArray(currentCards)).toBeTruthy()
+    })
+
+    it('should create a new card regardless if title already exists', () => {
+      expect(similarCards).toHaveLength(2)
+      expect(similarCards[0].title).toEqual(similarCards[1].title)
+    })
+
+    it('should not create card if title and id exists', () => {
+      expect(similarCards[0].id).not.toEqual(similarCards[1].id)
+      expect(currentCards).toHaveLength(4)
+    })
+  })
+
+  describe('createNewLine', () => {
+    const newLine = createNewLine(currentState.lines, { title: TEST_STRING_1, id: 1 }, bookId)
+
+    it('should return the line id', () => {
+      expect(typeof newLine).toBe('number')
```

Comment:
Does the line id change between test runs?
If not, perhaps we should check that it's a specific number.